File: recursion/ExpoNential.java
package recursion;

public class ExpoNential{
    public static void main(String[] args) {
        System.out.println(Raise(5, 0));
    }
    static int Raise(int base, int exp){
        if (exp == 0) {
            return 1;
        } else {
            return base * Raise(base, exp - 1);
        }
    }
}

File: recursion/exponential.py
# most efficient recursion

def raise_to_power(base, exp):
    if exp == 0:
        return 1
    elif exp < 0:
        return 1 / raise_to_power(base, -exp)
    else:
        half = raise_to_power(base, exp // 2)
        if exp % 2 == 0:
            return half * half
        else:
            return base * half * half

print(raise_to_power(5, 6))